service/history/ndc/history_replicator_test.go

```diff
@@ -0,0 +1,378 @@
+			mockExecutionMutableState := execution.NewMockMutableState(ctrl)
+			testReplicatorImpl.newBranchManager(mockExecutionContext, mockExecutionMutableState, log.NewNoop())
+		})
+	})
+}
+
+func TestNewHistoryReplicator_newConflictResolver(t *testing.T) {
+	assert.NotPanics(t, func() {
+		ctrl := gomock.NewController(t)
+
+		mockShard := shard.NewMockContext(ctrl)
+		mockShard.EXPECT().GetConfig().Return(&config.Config{
+			NumberOfShards:           0,
+			IsAdvancedVisConfigExist: false,
+			MaxResponseSize:          0,
+			HistoryCacheInitialSize:  dynamicconfig.GetIntPropertyFn(10),
+			HistoryCacheMaxSize:      dynamicconfig.GetIntPropertyFn(10),
+			HistoryCacheTTL:          dynamicconfig.GetDurationPropertyFn(10),
+			HostName:                 "test-host",
+		}).Times(2)
+
+		// before going into NewHistoryReplicator
+		mockExecutionManager := persistence.NewMockExecutionManager(ctrl)
+		mockShard.EXPECT().GetExecutionManager().Return(mockExecutionManager).Times(1)
+		mockShard.EXPECT().GetLogger().Return(log.NewNoop()).Times(1)
+		mockShard.EXPECT().GetMetricsClient().Return(nil).Times(3)
+
+		testExecutionCache := execution.NewCache(mockShard)
+		mockEventsReapplier := NewMockEventsReapplier(ctrl)
+
+		// going into NewHistoryReplicator -> newTransactionManager()
+		mockShard.EXPECT().GetClusterMetadata().Return(cluster.Metadata{}).Times(3)
+		mockHistoryManager := persistence.NewMockHistoryManager(ctrl)
+		mockShard.EXPECT().GetHistoryManager().Return(mockHistoryManager).Times(4)
+
+		mockHistoryResource := resource.NewMockResource(ctrl)
+		mockShard.EXPECT().GetService().Return(mockHistoryResource).Times(3)
+		mockPayloadSerializer := persistence.NewMockPayloadSerializer(ctrl)
+		mockHistoryResource.EXPECT().GetPayloadSerializer().Return(mockPayloadSerializer).Times(1)
+
+		// going into NewHistoryReplicator -> newTransactionManager -> reset.NewWorkflowResetter
+		mockDomainCache := cache.NewMockDomainCache(ctrl)
+		mockShard.EXPECT().GetDomainCache().Return(mockDomainCache).Times(4)
+
+		// going back to NewHistoryReplicator
+		mockHistoryResource.EXPECT().GetClusterMetadata().Return(cluster.Metadata{}).Times(2)
+
+		testReplicator := NewHistoryReplicator(mockShard, testExecutionCache, mockEventsReapplier, log.NewNoop())
+		testReplicatorImpl := testReplicator.(*historyReplicatorImpl)
+
+		// test newConflictResolver function in history replicator
+		assert.NotPanics(t, func() {
+			mockEventsCache := events.NewMockCache(ctrl)
+			mockShard.EXPECT().GetEventsCache().Return(mockEventsCache).Times(1)
+			mockShard.EXPECT().GetShardID().Return(0).Times(1)
+
+			mockExecutionContext := execution.NewMockContext(ctrl)
+			mockExecutionMutableState := execution.NewMockMutableState(ctrl)
+			testReplicatorImpl.newConflictResolver(mockExecutionContext, mockExecutionMutableState, log.NewNoop())
+		})
+	})
+}
+
+func TestNewHistoryReplicator_newWorkflowResetter(t *testing.T) {
+	assert.NotPanics(t, func() {
```

Comment:
we don't need these `assert.NotPanics` wrappers. if it panics test will fail anyway